fastlane/lanes/localization.rb

```diff
@@ -223,50 +266,80 @@
+      locales: WORDPRESS_METADATA_GLOTPRESS_LOCALE_CODES,
+      commit_message: 'Update WordPress metadata translations'
     )
   end
 
   # Downloads the localized metadata (for App Store Connect) from GlotPress for the Jetpack app
   #
   desc 'Downloads the localized metadata (for App Store Connect) from GlotPress for the Jetpack app'
   lane :download_jetpack_localized_app_store_metadata do
-    # @FIXME: Replace this whole lane with a call to the future replacement of `gp_downloadmetadata` once it's implemented in the release-toolkit (see paaHJt-31O-p2).
-
-    # No need to `cd` into `fastlane` because of how Fastlane manages its paths internally.
-    sh './download_metadata.swift jetpack'
+    metadata_directory = File.join(PROJECT_ROOT_FOLDER, 'fastlane', 'jetpack_metadata')
 
-    # @TODO: Make the `fastlane/jetpack_metadata/en-US/release_notes.txt` path be the source of truth for the original copies in the future.
+    # FIXME: We should make the `fastlane/jetpack_metadata/default/release_notes.txt` path be the source of truth for the original copies in the future.
     # (will require changes in the `update_appstore_strings` lane, the Release Scenario, the MC tool to generate the announcement post…)
     #
-    # In the meantime, since GlotPress doesn't have the `en-US` notes because those are the ones used as originals, just copy the file to the right place for `deliver` to find
-    metadata_directory = File.join(PROJECT_ROOT_FOLDER, 'fastlane', 'jetpack_metadata')
+    # In the meantime, just copy the file to the right place for `deliver` to find, for the `default` pseudo-locale which is used as fallback
     release_notes_source = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'Jetpack', 'Resources', 'release_notes.txt')
-    FileUtils.cp(release_notes_source, File.join(metadata_directory, 'en-US', 'release_notes.txt'))
+    FileUtils.cp(release_notes_source, File.join(metadata_directory, 'default', 'release_notes.txt'))
+
+    # Download metadata translations from GlotPress
+    download_localized_app_store_metadata(
+      glotpress_project_url: GLOTPRESS_JETPACK_METADATA_PROJECT_URL,
+      locales: JETPACK_METADATA_GLOTPRESS_LOCALE_CODES,
+      metadata_directory: metadata_directory,
+      commit_message: 'Update Jetpack metadata translations'
+    )
+  end
 
-    metadata_path = File.join(metadata_directory, '**', '*.txt')
-    git_add(path: metadata_path, shell_escape: true)
+  def download_localized_app_store_metadata(glotpress_project_url:, locales:, metadata_directory:, commit_message:)
+    # FIXME: Replace this with a call to the future replacement of `gp_downloadmetadata` once it's implemented in the release-toolkit (see paaHJt-31O-p2).
+
+    locales_map = GLOTPRESS_TO_ASC_METADATA_LOCALE_CODES.slice(*locales)    
+    what_new_key = "v#{ios_get_app_version}-whats-new"
+    gp_downloadmetadata(
+      project_url: glotpress_project_url,
+      target_files: APPSTORECONNECT_METADATA_KEYS_TO_FILES_MAP,
```

Comment:
😱  Whoops, just noticed an issue there (introduced after refactoring the list of target files into a constant during my work):

 - `what_new_key` (which should be `whats_new_key`, really, eh) is defined on line 313 but never used
 - `APPSTORECONNECT_METADATA_KEYS_TO_FILES_MAP` is now a constant, which [uses a hardcoded `what_new_key` key (verbatim)](https://github.com/wordpress-mobile/WordPress-iOS/pull/18439/files#diff-ad77c85d7eb3034dec610211cb1747f7cac134e425448b1dcc246d71df3e8639R90), instead of relying on the variable defined on line 313 here to know which key to use.


While we've talked (in our respective RFCs for the refactor of the `.po` files and metadata handling) about unifying the key for the release notes, and could even consider using a static, invariant key name (rather than one that contains the version in its key) in the future… this is outside of the scope of this PR (and would fail to find the release notes for 19.7 anyway, given translations for those use the `v19.7-whats-new` key, not a key like `what_new_key`…)

We'll have to fix this before merging.

Reply:
Should be fixed via b61cfeda85891554910f43a69812ae3eb0b60bb7

I've tried by running `download_wordpress_localized_app_store_metadata` on top of this fix and indeed I'm getting a couple of translated `release_notes.txt` files now (at least the ones already translated in GlotPress) 🎉